src/cmdline_manager.ts

```diff
@@ -100,14 +109,19 @@ export class CommandLineManager implements Disposable {
             }
             case "cmdline_hide": {
                 logger.debug(`cmdline_hide`);
-                this.state.ignoreHideEvent = true;
-                this.input.hide();
+                this.cmdlineHide();
                 break;
             }
         }
     }
 
-    private cmdlineShow = (content: string, firstc: string, prompt: string): void => {
+    private cmdlineShow = (content: string, firstc: string, prompt: string, level: number): void => {
+        if (!this.isVisible()) {
+            // Reset the state if this is a new dialog
+            this.reset();
```

Comment:
I moved this here because realistically, it's most important that _new_ dialogs have the proper state, rather than clearing it on close. It's a bit of a style thing, but I think that by doing this, we're also preventing bugs that crop up as a result of `onChange` events that are fired after we already closed the dialog